src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -642,23 +643,55 @@ SDCPUThread::populateDependencies(shared_ptr<InflightInst> inst_ptr)
     // }
 
     /*
-    * Note: An assumption is being made that inst_ptr is at the end of the
-    *       inflightInsts buffer.
-    */
+     * Note: An assumption is being made that inst_ptr is at the end of the
+     *       inflightInsts buffer.
+     */
 
     const StaticInstPtr static_inst = inst_ptr->staticInst();
 
     // BEGIN ISA explicit serialization
 
-    const bool we_ser_before = static_inst->isSerializeBefore();
+#define SDCPU_STRICT_SERIALIZATION
```

Comment:
I don't like using ifdefs if we can avoid it. Should this be a SimObject parameter, maybe? That could make sense while we're testing things, at least.

Reply:
Oh that's a good idea, didn't think of making it configurable that way...

Reply:
Just to follow up on this. You'll want to add something like the following line to the SimObject description file (SimpleDataflowCPU.py).

`strict_serialization = Param.Bool(True, "Controls behavior of serializing instructions")`

Then, in the constructor of the C++ class `SimpleDataflowCPU`, you can pass the parameter to the thread. You can access the parameter with `params->strict_serialization`.

Reply:
I may or may not have just woken up :b, so this will be something I'll be in to work on later, thanks